### ISSUE TITLE
Feature Request: tail output of `deis run`

### ISSUE THREAD

**Issue by jwaldrip** [deis/deis#3549]
I see that when running on off command with deis-run that the output is not streamed. It would be great if it were.

**Comment by bacongobbler** [comment]
related: #117

this is due to a limitation of WSGI; it cannot handle long-running responses so we just dump the output at the end of the run operation. With support for long-running requests by hooking into something like gevent, we should be able to support tailing the output of the ssh tunnel in the response.

Another idea is to attach the run to deis-logger and expose a session in the response, similar to heroku's logplex tail API: https://github.com/heroku/logplex/blob/master/doc/logplex-tail.png. From there, the client can read the URL and connect to the session.

**Comment by jwaldrip** [comment]
Is this a limitation of Python? Have you looked at Go for components related to keeping a long running open socket? IMHO This is something that Go thrives at.

**Comment by bacongobbler** [comment]
> Is this a limitation of Python? 

Lack of support for long-running requests is a limitation of WSGI, not necessarily a limitation of Python itself. :)

> Have you looked at Go for components related to keeping a long running open socket?

*Disclaimer: all thoughts laid out here are my own opinion*

That's why the logger's built in Go! Go's great for low-level systems programming such as a log server, but it **sucks** for handling API requests, authn/authz integrations, serialization/de-serialization of web requests, crafting responses, database interactions, etc. Django excels at all those points. It's a kickass web framework for crafting APIs. There's just no equivalent of Django in Go that meets our needs, and rightfully so, because Go was never meant to solve that problem domain. :)

There are evented solutions like python-twisted, gunicorn or raw gevent which can address this specific issue in the controller, but the more *maintainable* and *extendable* solution would be to roll out logging to a component which is built to host and serve logs, and just let the controller hand off the work to that component. This is the micro-services architecture in play. This way we can knock off more features by recycling the same endpoints while keeping the scope of the controller focused and the line count low. For example, [`deis logs -f`](465) has been a long-running feature request which can take advantage of a sessions API from the logger!

The thought of pivoting 2 years of code over from Python to Go scares me. I can only expect it to be fraught with compatibility issues and limitations which we've come to rely on. :fearful:

**Comment by jwaldrip** [comment]
@bacongobbler agree to disagree I guess. I have experience building great API's in go. For building an app with a UI component I would agree. But serialization/deserialization of data. Not difficult in go at all. This is coming from a veteran ruby/rails developer. And in the past I made your same argument. That is until I spent the time learning and getting to know go for apis. All in all I write and structure my go like a rails app. Routes, models, controllers the works. Its faster and it has wicked support for things like streaming. Next time you are in Boulder/Denver, I will buy you a beer and we can talk about it more :-).
